flow-nodes/api-builder-plugin-format-date/src/actions.js

```diff
@@ -21,25 +21,29 @@ const moment = require('moment');
  *	 does not define "next", the first defined output).
  */
 function formatDate(params) {
-	const { date, format } = params;
+  const { date, format, offset } = params;
 
-	if (!date) {
-		throw new Error('Missing required parameter: date');
-	}
-	if (!format) {
-		throw new Error('Missing required parameter: format');
-	}
+  // Defaults to now
+  const momentObj = moment(date);
 
-	// There is no sanity check on the formats in Moment.js, so an invalid format could be
-	// specified. As this is a wrapper, I won't be introducing additional format validation.
-	const formattedDate = moment(date).format(format);
+  if (!momentObj.isValid()) {
+ 		throw new Error(`Invalid date: ${date}`);
+ 	}
 
-	// Moment doesn't throw an error if the passed in date is invalid it just returns 'Invalid date'
-	if (formattedDate === 'Invalid date') {
-		throw new Error(`Invalid date: '${date}`);
-	}
-	return formattedDate;
-}
+  if (offset) {
+    const pattern = new RegExp('^[+-][0-1][0-9]:[0-5][0-9]$');
```

Comment:
Not a suggestion to change, but it can be quite expensive to the cpu to keep creating regexes. You'll notice in some cases we pull this out and declare it globally so that the regex is reused. The downside of this is that it will be stuck in memory the whole time. In this case, if we declare it at the top of the file, it will be in memory for everyone that installs the plugin even if/while they aren't using it.

Since it's not likely for users to be calling this flow node many times in a short period of time, so it's not going to be that bad to declare it where you have now. But for the future, it's good to be aware of this and have a think where is the best place to declare the regex.